gui/mozregui/report.py

```diff
@@ -73,22 +82,53 @@ def clear(self):
 
     @Slot(object)
     def attach_bisector(self, bisector):
-        slots = ('step_started', 'step_build_found', 'step_testing',
-                 'step_finished', 'started', 'finished')
+        bisector_slots = ['step_started',
+                          'step_build_found',
+                          'step_testing',
+                          'step_finished',
+                          'started',
+                          'finished']
+        downloader_slots = ['download_progress']
+
         if self.bisector:
-            # disconnect previous bisector
-            for name in slots:
-                signal = getattr(self.bisector, name)
-                slot = getattr(self, name)
-                signal.disconnect(slot)
+            _bulk_action_slots('disconnect',
+                               bisector_slots,
+                               self.bisector,
+                               self)
+            _bulk_action_slots('disconnect',
+                               downloader_slots,
+                               self.bisector.download_manager,
+                               self)
+
         if bisector:
-            # connect the new bisector
-            for name in slots:
-                signal = getattr(bisector, name)
-                slot = getattr(self, name)
-                signal.connect(slot)
+            _bulk_action_slots('connect',
+                               bisector_slots,
+                               bisector,
+                               self)
+            _bulk_action_slots('connect',
+                               downloader_slots,
+                               bisector.download_manager,
+                               self)
+
         self.bisector = bisector
 
+    @Slot(object, int, int)
+    def download_progress(self, dl, current, total):
+        item = self.items[-1]
+        item.progress = (current * 100) / total
+        self.update_item(item)
+
+    def is_downloading(self):
```

Comment:
This should go on the Item class I think (the item hold the downloading attribute).
I think you could do something like:
```
class Item(object):
    def is_downloading(self):
        return False

class StepItem(Item):
    def is_downloading(self):
        return self.downloading
```